### PR TITLE
Fixes for building mystikos in a docker env

### LINKED ISSUE #779: Dockerfile is out of date.
Following the instruction from 

https://github.com/deislabs/mystikos/blob/main/docker/build-myst/README.md

I got the error information:

Collecting cryptography (from ansible==2.8.8->-r ./openenclave/scripts/ansible/requirements.txt (line 3))
  Downloading https://files.pythonhosted.org/packages/cc/98/8a258ab4787e6f835d350639792527d2eb7946ff9fc0caca9c3f4cf5dcfe/cryptography-3.4.8.tar.gz (546kB)
    Complete output from command python setup.py egg_info:

            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:

            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ws9iz7uj/cryptography/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-ws9iz7uj/cryptography/
The command '/bin/sh -c ./openenclave/scripts/ansible/install-ansible.sh' returned a non-zero code: 1

### PR DESCRIPTION
**Pull request by vtikoo**
Fixes #779.

Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

**Comment by vtikoo** [comment]
Currently failing with - 
```
DKMS: add completed.
Error! Your kernel headers for kernel 5.4.0-1056-azure cannot be found.
Please install the linux-headers-5.4.0-1056-azure package,
or use the --kernelsourcedir option to tell DKMS where it's located
Makefile:23: recipe for target 'install_sgx_driver' failed
make[3]: *** [install_sgx_driver] Error 3
make[3]: Leaving directory '/src/prereqs'
Makefile:6: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/src/prereqs'
/src/rules.mak:55: recipe for target 'dirs' failed
make[1]: *** [dirs] Error 2
make[1]: Leaving directory '/src'
Makefile:11: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by vtikoo** [comment]
After adding `--device /dev/sgx:/dev/sgx` to the docker run command, it progresses further. Now failing with - 
```
make[5]: Entering directory '/src/third_party/openenclave/openenclave/build'
make[5]: /home/vitikoo/.local/lib/python2.7/site-packages/cmake/data/bin/cmake: Command not found
Makefile:1466: recipe for target 'cmake_check_build_system' failed
make[5]: *** [cmake_check_build_system] Error 127
make[5]: Leaving directory '/src/third_party/openenclave/openenclave/build'
Makefile:86: recipe for target 'install_oe' failed
make[4]: *** [install_oe] Error 2
make[4]: Leaving directory '/src/third_party/openenclave'
Makefile:29: recipe for target '/src/build/include/openenclave' failed
make[3]: *** [/src/build/include/openenclave] Error 2
make[3]: Leaving directory '/src/third_party/openenclave'
Makefile:11: recipe for target 'openenclave' failed
make[2]: *** [openenclave] Error 2
make[2]: Leaving directory '/src/third_party'
/src/rules.mak:55: recipe for target 'dirs' failed
make[1]: *** [dirs] Error 2
make[1]: Leaving directory '/src'
Makefile:11: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by jxyang** [comment]
Could you also clean up build-myst.sh? It should be `make distclean` as you mentioned. Also some comments are outdated.

File: docker/build-myst/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-ADD . /src
-WORKDIR /src 
-
 COPY ./build-myst.sh /build-myst.sh
 RUN /bin/chmod 777 /build-myst.sh
 
-RUN apt-get update
-RUN apt-get -y install git
+RUN apt-get update && \
+    apt-get -y install git sudo wget
```
Comment (jxyang) [inline docker/build-myst/Dockerfile:7]: Why do we need `sudo`?

Comment (vtikoo) [reply docker/build-myst/Dockerfile:7]: We have `sudo` in a couple of places in our Makefiles.

During clean targets -
https://github.com/deislabs/mystikos/blob/main/rules.mak#L59
https://github.com/deislabs/mystikos/blob/main/Makefile#L88

prereqs installation - 
https://github.com/deislabs/mystikos/blob/main/prereqs/Makefile#L9-L24

Comment (vtikoo) [reply docker/build-myst/Dockerfile:7]: The prereqs installation could potentially be moved within the dockerfile.

**Review by jxyang — APPROVED** [review]

**Pull request merged**